CodeStarterCamp_Week2/PlayLotto.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  PlayLotto.swift
+//  CodeStarterCamp_Week2
+//
+//  Created by 정선아 on 2022/04/22.
+//
+
+import Foundation
+
+func playLotto() {
+    pickLottoNumbers()
+    matchNumber(myLottoNumbers, lottoNumbers: lottoNumbers)
+    comfirmNumbers()
+}
+
+//찍은 번호
+let myLottoNumbers: [Int] = [4, 13, 24, 36, 38, 45]
+
+//로또 번호 배열
+var lottoNumbers: [Int] = []
+
+//로또 번호 랜덤 뽑기
+func pickLottoNumbers() {
+    while lottoNumbers.count < 6 {
+        let number = Int.random(in: 1...45)
+        if lottoNumbers.contains(number) == false {
+            lottoNumbers.append(number)
+        }
+    }
+}
+
+//찍은 번호와 로또 번호가 일치하면 들어가는 배열
+var matchedNumbers: [Int] = []
+
+func matchNumber(_ myLottoNumbers: [Int], lottoNumbers: [Int]) {
+    for number in myLottoNumbers {
+        if lottoNumbers.contains(number) {
+            matchedNumbers.append(number)
+        }
+    }
+}
+
+//matchedNumbers를 String타입으로 변환하는 과정
+func convertArray(from array: [Int]) -> [String] {
+    let convertArray = array.map({ (number: Int) -> String in return "\(number)"
+        return "\(number)"
+    })
+    return convertArray
+}
```

Comment:
이 부분 메서드 정말 멋진데요~!
리턴하는 게 `[String]`이 아니라, 여기서 메시지로 바로 내보낼 수 있게 모든 편집을 끝내서, `String` 으로 내보내는 게 훨씬 좋을 것 같아요. 😄 

그리고 메서드 이름은 기존 convertArray 에서 → convertToString 으로 바꿨어요. 왜냐면 결국 리턴되는 건 String 타입이니까요. ㅎㅎㅎ
무엇으로 부터 만들어낼 건지는 뒤에 따라오는 전치사 `from` 뒤에 들어갈 배열이겠죠! 👍🏻 

그리고 메서드 내부에서 임시 상수 네이밍할 때 제가 `As` 라는 전치사를 사용했는데요.
`To` 와 `As` 는 어떤 늬앙스 차이가 있을지 한 번 꼭!!! 고민해 보셨으면 좋겠어요! 🔥 
답을 알려드리진 않을테니, 솜이 고민해보시고 다시 코멘트 달아주세요. 😄 

그리고 출력할 때 숫자들 사이에 콤마(,) 뿐만 아니라, 콤마 뒤에 여백도 있어야 해요!
그래서 separator 또한 살짝 고쳤습니다~ ㅎㅎ

고차함수 map 뒤에 오는 저 `딸러($) 표시` 뭐지...? 싶으실 수 있지만
이건 나중에 알게되실 겁니다! 천천히 공부해요 ㅎㅎ

```suggestion
func convertToString(from array: [Int]) -> String {
    let numbersAsString = array.sorted().map { $0.description }
    let joinedMessage = numbersAsString.joined(separator: ", ")
    return joinedMessage
}
```

Reply:
```swift
func convertArray(from array: [Int]) -> [String] {
```
이 부분 저도 ```String 타입```으로 반환받고 싶어서 대괄호 표현 없이 적으니까 오류 메세지가 옆에 뜨더라고요. 그래서 정수 배열을 받아서 문자열로 변환하려 하는 거니까, 문자열도 배열로 만들어야 하는 걸까? 싶어서 대괄호를 삽입하니까 오류가 사라졌어요... 
뭔가 예거가 제안해주신 코드는 ```$0``` 역할 때문에(?) ```String 타입```으로도 변환되는 건가 궁금증이 생기네요! 
더 찾아보겠습니다✨✨

```as``` 와 ```to```의 차이... 갑자기 학창시절로 돌아가 전치사를 배우던 때가 생각났어요ㅎㅎ
~저는 수학이랑 영어를 제일 싫어했는데 지금은 둘 다 하게 되었네요 ㅎㅎㅠㅜ~
```to```는 어디서 어디로 이동하는 느낌의 전치사고, ```as```는 '~처럼' 이라는 전치사니까, ```numbersAsString```은 곧 숫자가 문자처럼 바뀐다는 뜻의 상수인 거 같아요! as의 뜻을 찾아봤는데, 여러개의 뜻 중에 '~처럼' 이 제일 적합할 거 같아요ㅎㅎ

수정했습니다:)

Reply:
솜이 저 메서드를 구현하실 때, 리턴 타입에 `String`을 하면 에러가 났던 이유는,
`map` 고차함수의 작동 방식이, Array 내부를 하나로 합쳐서, 뱉어내는 게 아니라
Array 내부의 원소들을 하나하나 `"\(number)"` 처리한 거에요.

즉, [1, 5, 7, 9, 11, 13] 라는 [Int] 가 들어가게 되면, map 처리를 끝내고 나올 때는
`["1", "5", "7", "9", "11", "13"]`가 되어서 나오는 거에요. 즉, `[String]` 인 거죠 ㅎㅎ

아직은 이해가 잘 안 되더라도, 그냥 그런 게 있나보다 하고 넘어가셔도 괜찮아요!
딸러 표시도 마찬가지구요 ㅎㅎㅎ 궁금하시면 [Shorthand Argument Names](https://docs.swift.org/swift-book/LanguageGuide/Closures.html) 찾아보시면 됩니다!

![image](https://user-images.githubusercontent.com/71127966/164750552-4ed7cc91-ca35-4032-bc20-f4a720069a72.png)

Reply:
제가 전치사 `To` 하고 `As`에 대해 질문드렸는데요. 😄 
제가 생각하는 것과 똑같이 대답해주셨어요! 👍🏻 

아래 메서드를 보면, 메서드 이름에는 `To`가 쓰이고
내부에서 sorted() 처리하고, map 고차함수를 통해 [String] 으로 변환된 array 를 잠시 저장해두는 `지역 상수` 네이밍에는
`As`를 썼죠.

사실 메서드 네이밍을 `convertAsString` 으로 지었어도, 솔직히 늬앙스 차이는 거의 없는 것 같긴 해요 😅 
이렇게 애매할 때는 `숙어 표현`을 참고하면 좋아요!

"convert A as B" 하고 "convert A to B" 각각 구글링을 해보는 거죠.
그래서 어떤 표현이 더 많이 쓰이는지 확인한 후, 어떤 `전치사`를 쓸 건지 고르면 됩니다!

- 방금 알았는데, 정확한 숙어 표현은 `convert A into B`군요. ㅎㅎㅎ
그러면 `convertIntoString(from ~)` 으로 지었어도 괜찮았을 거 같아요!
이런 식으로 숙어 표현을 참고하면 전치사 때문에 골치 아플 일이 줄어듭니다. ☺️ 

```swift
func convertToString(from array: [Int]) -> String {
    let numbersAsString = array.sorted().map { $0.description }
    let joinedMessage = numbersAsString.joined(separator: ", ")
    return joinedMessage
}
```